pkg/certificate/manager.go

```diff
@@ -128,12 +128,24 @@ func (m *Manager) start(ctx context.Context, mrcClient MRCClient) error {
 	return nil
 }
 
-// GetTrustDomain returns the trust domain from the configured signingkey issuer.
+// TrustDomain is used to hold the current certificate information about the trust domain
+type TrustDomain struct {
+	Signing    string
+	Validating string
+}
+
+// AreDifferent returns true if the signing and validating trust domains are different
+func (td TrustDomain) AreDifferent() bool {
+	return td.Signing != td.Validating
+}
+
+// GetTrustDomains returns the trust domains from the configured issuers.
 // Note that the CRD uses a default, so this value will always be set.
-func (m *Manager) GetTrustDomain() string {
+// It is up to the caller to determine if the signing and validating trust domains are different
+func (m *Manager) GetTrustDomains() TrustDomain {
```

Comment:
not sure I follow, in what sense?

Reply:
This gets called a few times in the path for generating an envoy config, so it's order magnitude on # of proxies over frequency of proxy broadcasts.

Technically also impacted by # of traffic targets, so in a fully connected mesh in non-permissive mode it can be substantially higher (O(NM)).